### PR TITLE
dataclients/kubernetes: add tests for traffic split with custom route

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
This change adds tests to demonstrate the problem described in #2268

For https://github.com/zalando/skipper/issues/2268

### CODE REVIEW

File: dataclients/kubernetes/testdata/ingressV1/traffic/issue2268.eskip [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Custom route has the same number of predicates as the service_1 route
+// which causes undefined behavior: a request with foo cookie might be routed to any of them.
+kube_ns__issue2268_custom_0__test_example_org_____: Host("^(test[.]example[.]org[.]?(:[0-9]+)?)$") && PathSubtree("/") && Cookie("foo", "^") -> status(500) -> <shunt>;
+kube_ns__issue2268__test_example_org_____service_1: Host("^(test[.]example[.]org[.]?(:[0-9]+)?)$") && PathSubtree("/") && Traffic(0.8)       -> "http://42.0.0.1:8080";
+kube_ns__issue2268__test_example_org_____service_2: Host("^(test[.]example[.]org[.]?(:[0-9]+)?)$") && PathSubtree("/")                       -> "http://42.0.0.2:8080";
```
Comment (RomanZavodskikh) [inline dataclients/kubernetes/testdata/ingressV1/traffic/issue2268.eskip:5]: Shouldn't we have `True()` predicates here?

Comment (AlexanderYastrebov) [reply dataclients/kubernetes/testdata/ingressV1/traffic/issue2268.eskip:5]: Current implementation adds True() predicates only when there are more than two services, see #2303

Comment (szuecs) [reply dataclients/kubernetes/testdata/ingressV1/traffic/issue2268.eskip:5]: The question is what do you expect to happen for requests with cookie `foo`?
I would guess: always match the custom route
What if the custom route use the same svc backend?

File: dataclients/kubernetes/testdata/routegroups/traffic/issue2268.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      serviceName: service-1
+      servicePort: 8080
+    - name: service-2
+      type: service
+      serviceName: service-2
+      servicePort: 8080
+    - name: shunt
+      type: shunt
+  routes:
+    # custom route
```
Comment (szuecs) [inline dataclients/kubernetes/testdata/routegroups/traffic/issue2268.yaml:21]: how do you define `custom route` in this context?
1. a non service route (by backend type for example)
2. a route that is not part of the traffic switching

Comment (AlexanderYastrebov) [reply dataclients/kubernetes/testdata/routegroups/traffic/issue2268.yaml:21]: the 2.

File: dataclients/kubernetes/testdata/ingressV1/traffic/issue2268.eskip (hunk 1 above)
Comment (AlexanderYastrebov) [reply dataclients/kubernetes/testdata/ingressV1/traffic/issue2268.eskip:5]: I add this test to capture the current unexpected behavior.
I plan to use the same routegroup/ingress to show how TrafficSegment-based implementation solves this problem.

Comment (AlexanderYastrebov) [reply dataclients/kubernetes/testdata/ingressV1/traffic/issue2268.eskip:5]: > What if the custom route use the same svc backend?

It is not possible to use service backend with ingress.
The RouteGroup like this:
```yaml
  routes:
    # custom route
    - pathSubtree: /
      predicates:
        - Cookie("foo", "^")
      filters:
        - status(500)
      backends:
        - backendName: service-1
          weight: 80
        - backendName: service-2
          weight: 20
    # service route
    - pathSubtree: /
      backends:
        - backendName: service-1
          weight: 80
        - backendName: service-2
          weight: 20
```
would result in 
```
kube_rg__ns__issue2268__all__0_0: Host("^(test[.]example[.]org[.]?(:[0-9]+)?)$") && PathSubtree("/") && Cookie("foo", "^") && Traffic(0.8) -> status(500) -> "http://42.0.0.1:8080";
kube_rg__ns__issue2268__all__0_1: Host("^(test[.]example[.]org[.]?(:[0-9]+)?)$") && PathSubtree("/") && Cookie("foo", "^")                 -> status(500) -> "http://42.0.0.2:8080";
kube_rg__ns__issue2268__all__1_0: Host("^(test[.]example[.]org[.]?(:[0-9]+)?)$") && PathSubtree("/") && Traffic(0.8)                       -> "http://42.0.0.1:8080";
kube_rg__ns__issue2268__all__1_1: Host("^(test[.]example[.]org[.]?(:[0-9]+)?)$") && PathSubtree("/")                                       -> "http://42.0.0.2:8080";

```
I.e. the same problem.

Comment (AlexanderYastrebov) [reply dataclients/kubernetes/testdata/ingressV1/traffic/issue2268.eskip:5]: I can add a second routegroup case for "custom" route with traffic backend

**Comment by szuecs** [comment]
:+1:

File: dataclients/kubernetes/testdata/routegroups/traffic/issue2268.yaml (hunk 2 above)
Comment (AlexanderYastrebov) [reply dataclients/kubernetes/testdata/routegroups/traffic/issue2268.yaml:21]: The route is defined by its predicates, filters and backends. "custom" is just a name.

**Pull request merged**